### PR TITLE
Parse Redis cluster and sentinel urls

### LINKED ISSUE #554: Too Many Colons error when using Redis session store
## Expected Behavior

I am trying to enable redis session storage when using oauth2-proxy v5.1.1. I have the following flags in place:

```
--redis-use-cluster=true \
--redis-cluster-connection-urls=redis://redis-0:6379 \
--session-store-type=redis
```

I'm running kubernetes and the `redis-0` resolves properly. I expect to be able to go through the normal auth flow with these flags set. 

This 

## Current Behavior

When I auth against oauth2-proxy with these flags in place I see the following in the logs:

```
oauth2-proxy 127.0.0.1 - btroutwine@goodwatercap.com [2020/05/14 00:25:49] [AuthSuccess] Authenticated via OAuth2: Session{email:btroutwine@goodwatercap.com user:106428469046097333554 PreferredUsername: token:true id_token:true created:2020-05-14 00:25:49.180657438 +0000 UTC m=+39.577027380 expires:2020-05-14 01:25:48 +0000 UTC refresh_token:true}                                                                                                              
oauth2-proxy [2020/05/14 00:25:49] [oauthproxy.go:792] 127.0.0.1:49444 ("127.0.0.1") dial tcp: address redis://redis-0:6379: too many colons in address
```

This results in an HTTP 500 from oauth2-proxy. 

## Possible Solution

Unsure. The error message "too many colons in address" comes up often after a quick search, such as https://github.com/go-redis/redis/issues/864. 

## Steps to Reproduce (for bugs)

1. Posses a functional oauth2-proxy setup using cookie session store.
2. Configure this oauth2-proxy to use redis session store. 
3. Re-authenticate and observe the logs. 

I have confirmed this is reproducible with an IP address, without the `redis://` and I tried without an explicit port but this appears to be obligatory despite the documented optionality. 

## Context

I am unable to use redis for session storage. 

## Your Environment

I am running in GKE with oauth2-proxy behind an nginx-ingress. We have found that refresh cookies are too large for our current nginx-ingress header max sizes and wanted to avoid increasing this maximum by pushing session storage to redis and away from cookies. 

- Version used: v5.1.1

### PR DESCRIPTION
**Pull request by amnay-mo**
## Description

Properly parse Redis sentinel and cluster URLs

## Motivation and Context

-  Fixes #554
- redis.Options expects a list of `host:port` addresses, instead we were passing it a list of `redis://host:port` urls

## How Has This Been Tested?

I tested with a locally run redis cluster and sentinel

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: pkg/sessions/redis/redis_store.go [hunk 1]
```diff
@@ -60,16 +60,24 @@ func newRedisCmdable(opts options.RedisStoreOptions) (Client, error) {
 	}
 
 	if opts.UseSentinel {
+		addrs, err := parseRedisURLs(opts.SentinelConnectionURLs)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse redis urls: %s", err)
```
Comment (JoelSpeed) [inline pkg/sessions/redis/redis_store.go:65]: Please use `%v` for error types

**Comment by amnay-mo** [comment]
Yes, I have tested this using a locally run redis cluster and sentinel
Testing this would require spawning multiple redis instances, I don't know if it is worth the trouble

**Comment by JoelSpeed** [comment]
Doesn't look like miniredis supports cluster so you're right, might be a bit harder to set up tests.

Perhaps we merge as is and create an issue to investigate tests afterwards

**Comment by amnay-mo** [comment]
I found this: https://github.com/Bose/minisentinel
I can add tests for the redis sentinel usecase for a start

**Comment by amnay-mo** [comment]
I have added tests for the redis session store :)

File: pkg/sessions/redis/redis_store.go [hunk 2]
```diff
@@ -108,6 +116,20 @@ func newRedisCmdable(opts options.RedisStoreOptions) (Client, error) {
...
 }
 
+// parseRedisURLs parses a list of redis urls and returns a list
+// of addresses in the form of host:port that can be used to connnect to Redis
+func parseRedisURLs(urls []string) ([]string, error) {
+	addrs := []string{}
+	for _, u := range urls {
+		parsedURL, err := redis.ParseURL(u)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse redis url: %s", err)
```
Comment (JoelSpeed) [inline pkg/sessions/redis/redis_store.go:126]: It's idiomatic to use %v for errors 🙂 
```suggestion
			return nil, fmt.Errorf("unable to parse redis url: %v", err)
```

**Pull request merged**